lib/model-builder.js

```diff
@@ -308,7 +308,9 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
   // Add the id property
   if (idInjection) {
     // Set up the id property
-    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true});
+    // updateOnly property is added to indicate that this property will appear in the model for update/updateorcreate
+    // operations but and not for create operation.
+    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true, updateOnly: true});
```

Comment:
@bajtos I saw lot of tests failed in CI in downstream component 'loopback' if calculated forceId is set back into model class's settings[]  in mode-builder.define (..) method instead of datasource.setupDataAccess(). See in this commit 
 https://github.com/strongloop/loopback-datasource-juggler/pull/1453/commits/bb9be2d4c50d2ab99aecce12655db44cc6779a56#diff-546e03515f85748fe5b4d0db11cd81b7R355

I am bit puzzled what's the difference between setting it in model-builder.js vs datasource.js. 

TO avoid the CI failues, I reverted the refactoring of model-builder.js and datasource.js with my latest commit. PTAL.